src/devices/RotaryEncoder/QuadratureRotaryEncoder.cs

```diff
@@ -50,13 +58,15 @@ public class QuadratureRotaryEncoder
         /// <param name="pinB">Pin B that is connected to the rotary encoder. Sometimes called data</param>
         /// <param name="edges">The pin event types to 'listen' for.</param>
         /// <param name="pulsesPerRotation">The number of pulses to be received for every full rotation of the encoder.</param>
-        public QuadratureRotaryEncoder(GpioController controller, int pinA, int pinB, PinEventTypes edges, int pulsesPerRotation)
+        /// <param name="shouldDispose">True to dispose the controller</param>
+        public QuadratureRotaryEncoder(GpioController? controller, int pinA, int pinB, PinEventTypes edges, int pulsesPerRotation, bool shouldDispose = true)
```

Comment:
in most of the bindings gpioController is after required args and it defaults to null so that you are not required to provide it if not needed.

Reply:
fixing as well